### PR TITLE
Update routing.py to handle circular references

### PR DESCRIPTION
**Pull request by JavierSanchezCastro**
From #11424

### CODE REVIEW

File: fastapi/routing.py [hunk 1]
```diff
@@ -1204,6 +1204,7 @@ def read_users():
         app.include_router(internal_router)
         ```
         """
+        assert id(self) != id(router), "Router cannot be the same as parent"
```
Comment (killjoy1221) [inline fastapi/routing.py:1207]: would `self is not router` not be sufficient?

Comment (JavierSanchezCastro) [reply fastapi/routing.py]: That's completely true, updated!

**Review by mattmess1221 — APPROVED** [review]

**Review by eljoserass — APPROVED** [review]
👍

**Comment by connesy** [comment]
Hi @JavierSanchezCastro 

Sorry for pinging you on an old PR, but was this PR ever merged into FastAPI master? I ran into this exact issue today, and couldn't find your `assert self is not router` in the source code.

**Comment by JavierSanchezCastro** [comment]
@connesy Hey, I think it got closed automatically because I made changes in my repo, I'll create another one!

**Pull request closed without merging**